### PR TITLE
OADP-4949: Errors in Opt-in & Opt-out

### PR DESCRIPTION
**Pull request by anarnold97**
### JIRA

* [OADP-4949](https://issues.redhat.com/browse/OADP-4949)

### Version(s):

* OCP 4.12 ++
Issue:

### PREVIEW

* [Advanced OADP topics](https://83008--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-advanced-topics.html)

### QE review:
- [X] [QE has approved this change](https://github.com/openshift/openshift-docs/pull/83008#issuecomment-2412828807).

### CODE REVIEW

File: modules/oadp-pod-volume-backup.adoc [hunk 1]
```diff
@@ -8,10 +8,11 @@
...
-* *Opt-out approach*: With the opt-out approach, you must actively specify that you want to exclude a volume from a backup. You do this by labelling each pod that contains a volume you do not want to back up with the name of the volume. When Velero finds a PV, it checks the pod that mounted the volume. If the pod is labelled with the volume's name, Velero does not back up the pod.
+* *Opt-in approach*: With the opt-in approach, volumes not opted-in use snapshot or DataMover, which is the default. Volumes, opted-in by using annotation, use FSB backup, Kopia or Restic. 
+
+* *Opt-out approach*: With the opt-out approach, volumes not opted out use filesystem backup, which is the default. Volumes, opted out by the addition of annotations, are backed up from a snapshot or Data Mover backup.
```
Comment (anarnold97) [inline modules/oadp-pod-volume-backup.adoc:15]:
```suggestion
* *Opt-out approach*: With the opt-out approach, volumes not opted-out use FSB, which is the default. Volumes, opted out by the addition of annotations, are backed up from a snapshot or Data Mover backup.
```

File: modules/oadp-pod-volume-backup.adoc [hunk 2]
```diff
@@ -8,10 +8,11 @@
...
+Velero supports two approaches for determining pod volumes. Use the opt-in or opt-out approach to allow Velero to decide between an FSB, a volume snapshot, or a Data Mover backup. 
 
-* *Opt-in approach*: The opt-in approach requires that you actively indicate that you want to include - _opt-in_ - a volume in a backup. You do this by labelling each pod that contains a volume to be backed up with the name of the volume. When Velero finds a persistent volume (PV), it checks the pod that mounted the volume. If the pod is labelled with the name of the volume, Velero backs up the pod.
-* *Opt-out approach*: With the opt-out approach, you must actively specify that you want to exclude a volume from a backup. You do this by labelling each pod that contains a volume you do not want to back up with the name of the volume. When Velero finds a PV, it checks the pod that mounted the volume. If the pod is labelled with the volume's name, Velero does not back up the pod.
+* *Opt-in approach*: With the opt-in approach, volumes not opted-in use snapshot or DataMover, which is the default. Volumes, opted-in by using annotation, use FSB backup, Kopia or Restic.
```
Comment (anarnold97) [inline modules/oadp-pod-volume-backup.adoc:13]:
```suggestion
* *Opt-in approach*: With the opt-in approach, volumes not opted-in use snapshot or DataMover, which is the default. Volumes, opted-in by using annotations, use FSB backup, Kopia or Restic. 
```

File: modules/oadp-pod-volume-backup.adoc [hunk 3]
```diff
@@ -8,10 +8,11 @@
...
+Velero supports two approaches for determining pod volumes. Use the opt-in or opt-out approach to allow Velero to decide between an FSB, a volume snapshot, or a Data Mover backup. 
 
-* *Opt-in approach*: The opt-in approach requires that you actively indicate that you want to include - _opt-in_ - a volume in a backup. You do this by labelling each pod that contains a volume to be backed up with the name of the volume. When Velero finds a persistent volume (PV), it checks the pod that mounted the volume. If the pod is labelled with the name of the volume, Velero backs up the pod.
-* *Opt-out approach*: With the opt-out approach, you must actively specify that you want to exclude a volume from a backup. You do this by labelling each pod that contains a volume you do not want to back up with the name of the volume. When Velero finds a PV, it checks the pod that mounted the volume. If the pod is labelled with the volume's name, Velero does not back up the pod.
+* *Opt-in approach*: With the opt-in approach, volumes not opted-in use snapshot or DataMover, which is the default. Volumes, opted-in by using annotations, use FSB backup, Kopia or Restic.
```
Comment (PrasadJoshi12) [inline modules/oadp-pod-volume-backup.adoc:13]: This looks confusing to me. Maybe we could change it to something like this. 
```suggestion
* *Opt-in approach*: With the opt-in approach, volumes will be backed up using FSB even though the backup spec is using snapshot or DataMover as the default option. Opted-in by using annotations, use Kopia or Restic. 
```

File: modules/oadp-pod-volume-backup.adoc [hunk 4]
```diff
@@ -8,10 +8,11 @@
...
 
-* *Opt-in approach*: The opt-in approach requires that you actively indicate that you want to include - _opt-in_ - a volume in a backup. You do this by labelling each pod that contains a volume to be backed up with the name of the volume. When Velero finds a persistent volume (PV), it checks the pod that mounted the volume. If the pod is labelled with the name of the volume, Velero backs up the pod.
-* *Opt-out approach*: With the opt-out approach, you must actively specify that you want to exclude a volume from a backup. You do this by labelling each pod that contains a volume you do not want to back up with the name of the volume. When Velero finds a PV, it checks the pod that mounted the volume. If the pod is labelled with the volume's name, Velero does not back up the pod.
+* *Opt-in approach*: With the opt-in approach, volumes not opted-in use snapshot or DataMover, which is the default. Volumes, opted-in by using annotations, use FSB backup, Kopia or Restic. 
+
+* *Opt-out approach*: With the opt-out approach, volumes not opted-out use FSB, which is the default. Volumes, opted out by the addition of annotations, are backed up from a snapshot or Data Mover backup.
```
Comment (PrasadJoshi12) [inline modules/oadp-pod-volume-backup.adoc:15]:
```suggestion
* *Opt-out approach*: With the opt-out approach, volumes are not backed up via FSB. Volumes, opted out by the addition of annotations, are backed up via a snapshot or Data Mover.
```

File: modules/oadp-pod-volume-backup.adoc [hunk 5]
```diff
@@ -8,10 +8,11 @@
...
 
-* *Opt-in approach*: The opt-in approach requires that you actively indicate that you want to include - _opt-in_ - a volume in a backup. You do this by labelling each pod that contains a volume to be backed up with the name of the volume. When Velero finds a persistent volume (PV), it checks the pod that mounted the volume. If the pod is labelled with the name of the volume, Velero backs up the pod.
-* *Opt-out approach*: With the opt-out approach, you must actively specify that you want to exclude a volume from a backup. You do this by labelling each pod that contains a volume you do not want to back up with the name of the volume. When Velero finds a PV, it checks the pod that mounted the volume. If the pod is labelled with the volume's name, Velero does not back up the pod.
+* *Opt-in approach*: With the opt-in approach, volumes will be backed up using FSB even though the backup spec is using snapshot or DataMover as the default option. Opted-in by using annotations, use Kopia or Restic. 
+
+* *Opt-out approach*: With the opt-out approach, volumes are not backed up via FSB. Volumes, opted out by the addition of annotations, are backed up via a snapshot or Data Mover.
```
Comment (anarnold97) [inline modules/oadp-pod-volume-backup.adoc:15]:
```suggestion
* *Opt-out approach*: With the opt-out approach, volumes are not backed up using a FSB. Volumes, opted out by the addition of annotations, are backed up via a snapshot or Data Mover.
```

File: modules/oadp-pod-volume-backup.adoc [hunk 6]
```diff
@@ -8,10 +8,11 @@
...
 
-* *Opt-in approach*: The opt-in approach requires that you actively indicate that you want to include - _opt-in_ - a volume in a backup. You do this by labelling each pod that contains a volume to be backed up with the name of the volume. When Velero finds a persistent volume (PV), it checks the pod that mounted the volume. If the pod is labelled with the name of the volume, Velero backs up the pod.
-* *Opt-out approach*: With the opt-out approach, you must actively specify that you want to exclude a volume from a backup. You do this by labelling each pod that contains a volume you do not want to back up with the name of the volume. When Velero finds a PV, it checks the pod that mounted the volume. If the pod is labelled with the volume's name, Velero does not back up the pod.
+* *Opt-in approach*: With the opt-in approach, volumes will be backed up using FSB even though the backup spec is using snapshot or DataMover as the default option. Opted-in by using annotations, use Kopia or Restic. 
+
+* *Opt-out approach*: With the opt-out approach, volumes are not backed up using a FSB. Volumes, opted out by the addition of annotations, are backed up via a snapshot or Data Mover.
```
Comment (anarnold97) [inline modules/oadp-pod-volume-backup.adoc:15]:
```suggestion
* *Opt-out approach*: With the opt-out approach, volumes are not backed up using FSB. Volumes, opted out by the addition of annotations, are backed up via a snapshot or Data Mover.
```

**Comment by anarnold97** [comment]
@PrasadJoshi12 && @sseago - please can i give you a nudge about this?

Thanks

File: modules/oadp-pod-volume-backup.adoc [hunk 7]
```diff
@@ -8,10 +8,11 @@
...
 
-* *Opt-in approach*: The opt-in approach requires that you actively indicate that you want to include - _opt-in_ - a volume in a backup. You do this by labelling each pod that contains a volume to be backed up with the name of the volume. When Velero finds a persistent volume (PV), it checks the pod that mounted the volume. If the pod is labelled with the name of the volume, Velero backs up the pod.
-* *Opt-out approach*: With the opt-out approach, you must actively specify that you want to exclude a volume from a backup. You do this by labelling each pod that contains a volume you do not want to back up with the name of the volume. When Velero finds a PV, it checks the pod that mounted the volume. If the pod is labelled with the volume's name, Velero does not back up the pod.
+* *Opt-in approach*: With the opt-in approach, volumes will be backed up using FSB even though the backup spec is using snapshot or DataMover as the default option. Opted-in by using annotations, use Kopia or Restic. 
+
+* *Opt-out approach*: With the opt-out approach, volumes are not backed up using FSB. Volumes, opted out by the addition of annotations, are backed up using a snapshot or Data Mover.
```
Comment (sseago) [inline modules/oadp-pod-volume-backup.adoc:15]: With the opt-out approach, volumes are backed up using FSB by default. Snapshots or Data Mover will be used on specific volumes opted-out by annotations.

**Comment by anarnold97** [comment]
@sseago - thanks

File: modules/oadp-pod-volume-backup.adoc [hunk 8]
```diff
@@ -8,10 +8,11 @@
...
 
-* *Opt-in approach*: The opt-in approach requires that you actively indicate that you want to include - _opt-in_ - a volume in a backup. You do this by labelling each pod that contains a volume to be backed up with the name of the volume. When Velero finds a persistent volume (PV), it checks the pod that mounted the volume. If the pod is labelled with the name of the volume, Velero backs up the pod.
-* *Opt-out approach*: With the opt-out approach, you must actively specify that you want to exclude a volume from a backup. You do this by labelling each pod that contains a volume you do not want to back up with the name of the volume. When Velero finds a PV, it checks the pod that mounted the volume. If the pod is labelled with the volume's name, Velero does not back up the pod.
+* *Opt-in approach*: With the opt-in approach, volumes are backed up using snapshot or Data Mover by default. FSB is used on on specific volumes opted-in by annotations.
+
+* *Opt-out approach*: With the opt-out approach, volumes are not backed up using FSB. Volumes, opted out by the addition of annotations, are backed up using a snapshot or Data Mover.
```
Comment (anarnold97) [inline modules/oadp-pod-volume-backup.adoc:15]:
```suggestion
* *Opt-out approach*: With the opt-out approach, volumes are backed up using FSB by default. Snapshots or Data Mover is used on specific volumes opted-out by annotations.
```

**Review by sseago — APPROVED** [review]

**Comment by anarnold97** [comment]
@PrasadJoshi12 - updated following Scott's suggestions. Are you ok for this PR to be merged, please? 

Thanks

**Comment by PrasadJoshi12** [comment]
LGTM

**Review by PrasadJoshi12 — APPROVED** [review]

**Comment by anarnold97** [comment]
Thanks @PrasadJoshi12 & @sseago

File: modules/oadp-pod-volume-backup.adoc [hunk 9]
```diff
@@ -8,10 +8,11 @@
 
 Before you start a backup operation by using File System Backup (FSB), you must specify which pods contain a volume that you want to back up. Velero refers to this process as "discovering" the appropriate pod volumes.
 
-Velero supports two approaches for determining pod volumes:
+Velero supports two approaches for determining pod volumes. Use the opt-in or opt-out approach to allow Velero to decide between an FSB, a volume snapshot, or a Data Mover backup.
```
Comment (abhatt-rh) [inline modules/oadp-pod-volume-backup.adoc:11]:
```suggestion
Velero supports two approaches for determining pod volumes. Use the opt-in or the opt-out approach to allow Velero to decide between an FSB, a volume snapshot, or a Data Mover backup. 
```

File: modules/oadp-pod-volume-backup.adoc [hunk 10]
```diff
@@ -8,10 +8,11 @@
...
+Velero supports two approaches for determining pod volumes. Use the opt-in or opt-out approach to allow Velero to decide between an FSB, a volume snapshot, or a Data Mover backup. 
 
-* *Opt-in approach*: The opt-in approach requires that you actively indicate that you want to include - _opt-in_ - a volume in a backup. You do this by labelling each pod that contains a volume to be backed up with the name of the volume. When Velero finds a persistent volume (PV), it checks the pod that mounted the volume. If the pod is labelled with the name of the volume, Velero backs up the pod.
-* *Opt-out approach*: With the opt-out approach, you must actively specify that you want to exclude a volume from a backup. You do this by labelling each pod that contains a volume you do not want to back up with the name of the volume. When Velero finds a PV, it checks the pod that mounted the volume. If the pod is labelled with the volume's name, Velero does not back up the pod.
+* *Opt-in approach*: With the opt-in approach, volumes are backed up using snapshot or Data Mover by default. FSB is used on on specific volumes opted-in by annotations.
```
Comment (abhatt-rh) [inline modules/oadp-pod-volume-backup.adoc:13]:
```suggestion
* *Opt-in approach*: With the opt-in approach, volumes are backed up using snapshot or Data Mover by default. FSB is used on specific volumes that are opted-in by annotations.
```

File: modules/oadp-pod-volume-backup.adoc [hunk 11]
```diff
@@ -8,10 +8,11 @@
...
+
+* *Opt-out approach*: With the opt-out approach, volumes are backed up using FSB by default. Snapshots or Data Mover is used on specific volumes opted-out by annotations.
```
Comment (abhatt-rh) [inline modules/oadp-pod-volume-backup.adoc:15]:
```suggestion
* *Opt-out approach*: With the opt-out approach, volumes are backed up using FSB by default. Snapshots or Data Mover is used on specific volumes that are opted-out by annotations.
```

**Review by abhatt-rh — COMMENTED** [review]
Some minor nits; rest LGTM!

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

**Comment by anarnold97** [comment]
@abhatt-rh - thanks, changes made

**Review by ShaunaDiaz — COMMENTED** [review]
/lgtm

**Pull request merged**